File: backend/src/main/scala/dev/cptlobster/appoint/orm/Availability.java
package dev.cptlobster.appoint.orm;

import jakarta.persistence.*;

import java.time.DayOfWeek;
import java.time.LocalTime;
import java.util.UUID;

/**
 * A weekly availability slot on an appointment schedule.
 */
@Entity
@Table(name = "availabilities")
public class Availability {
    @Id
    @GeneratedValue
    public UUID id;
    @Column(nullable = false)
    public DayOfWeek weekday;
    @Column(nullable = false)
    public LocalTime start;
    @Column(nullable = false)
    public LocalTime end;
}

File: backend/src/main/scala/dev/cptlobster/appoint/models/Availability.scala
package dev.cptlobster.appoint.models

import java.time.{DayOfWeek, LocalTime}
import java.util.UUID

case class Availability(id: UUID,
                        weekday: DayOfWeek,
                        startTime: LocalTime,
                        endTime: LocalTime) {

}
